src/ecdsa/keys.py

```diff
@@ -923,8 +947,48 @@ def from_der(cls, string, hashfunc=sha1):
         :return: Initialised SigningKey object
         :rtype: SigningKey
         """
-        string = normalise_bytes(string)
-        s, empty = der.remove_sequence(string)
+        s = normalise_bytes(string)
+        curve = None
+
+        # PKCS #8 has the algorithm identifier, including the curve name,
+        # before the actual key. Then it contains the key data within an
+        # octet string.
+        if pkcs8:
+            s, empty = der.remove_sequence(s)
+            if empty != b(""):
+                raise der.UnexpectedDER(
+                    "trailing junk after DER privkey: %s"
+                    % binascii.hexlify(empty)
+                )
+
+            version, s = der.remove_integer(s)
+            if version != 0 and version != 1:
+                raise der.UnexpectedDER(
+                    "expected '0' or '1' at start of DER privkey, got %d"
+                    % version
+                )
+
+            algorithm_identifier, s = der.remove_sequence(s)
+            algorithm_oid, algorithm_identifier = der.remove_object(
+                algorithm_identifier
+            )
+            curve_oid, empty = der.remove_object(algorithm_identifier)
+            curve = find_curve(curve_oid)
+
+            if algorithm_oid not in (oid_ecPublicKey, oid_ecDH, oid_ecMQV):
+                raise der.UnexpectedDER(
+                    "unexpected algorithm identifier '%s'" % algorithm_oid
+                )
+            if empty != b"":
+                raise der.UnexpectedDER(
+                    "unexpected data after algorithm identifier: %s"
+                    % binascii.hexlify(empty)
+                )
+
+            # We don't care about the optional fields after the key data.
```

Comment:
1. why?
2. shouldn't it be documented in the doc text?

Reply:
The optional fields are `attributes` (public key metadata, like certificates) and `publicKey`. They're show in the docstring; I can be explicit about why they're not needed.

Reply:
ok, please add comment what we're skipping here and a generic addition in doc string that only such and such fields are inspected (or such and such are completely ignored)